pkg/ds/daemon_set_test.go

```diff
@@ -309,7 +307,7 @@ func TestSchedule(t *testing.T) {
 	labeled = labeledPods(t, ds)
 	Assert(t).AreEqual(len(labeled), expectedNodes, "Expected no nodes to be unlabeled")
 
-	err = waitForPodsInIntent(consulStore, len(beforeDeletePods))
+	err = waitForPodsInIntent(consulStore, 12)
```

Comment:
as this code is written, 12 will be the right number (I'll explain why later), but I believe the test might still fail if the daemon set gets deleted (and stops replicating) before the pod makes it to all 12. So, maybe a `waitForPodsInIntent(consulStore, 12)` before the delete would be appropriate.

12 is currently the right number because the 11 nodeQuality=good nodes on which the pod was previously scheduled hit the `"Reality for this node matches this DS. No action required."` and therefore they don't get new pods in intent.

Changing the manifest or having something clear out reality will ensure that all 23 show up.